battle.go

```diff
@@ -125,24 +126,68 @@ func (b *Battle) SimulateRound() []Transaction {
 			// See: https://github.com/StevensSEC/pokemonbattlelib/wiki/Requirements#fight-using-a-move
 			modifier := uint(1) // TODO: damage multiplers
 			damage := (((2*uint(user.Level)/5)+2)*uint(user.Moves[t.Move].Power)*user.Stats[STAT_ATK]/receiver.Stats[STAT_DEF]/50 + 2) * modifier
-			transactions = append(transactions, DamageTransaction{
-				User:   &user,
-				Target: receiver,
-				Move:   user.Moves[t.Move],
-				Damage: damage,
+			queue = append(queue, DamageTransaction{
+				User:            &user,
+				Target:          receiver,
+				TargetParty:     target.party,
+				TargetPartySlot: target.partySlot,
+				Move:            user.Moves[t.Move],
+				Damage:          damage,
 			})
 		default:
 			log.Panicf("Unknown turn of type %v", t)
 		}
-	}
 
-	// process transations
-	for _, transaction := range transactions {
-		switch t := transaction.(type) {
-		case DamageTransaction:
-			(*t.Target).CurrentHP -= t.Damage
+		// process transations for this turn
+		for len(queue) > 0 {
```

Comment:
It might be worth creating a `TransactionQueue` in the future to improve readability of code written here

Reply:
I mean it's not _that_ hard to read is it?

Reply:
no its not, but I might not hurt as we continue to extend its functionality